kombu/transport/virtual/exchange.py

```diff
@@ -51,8 +52,14 @@ class DirectExchange(ExchangeType):
     type = 'direct'
 
     def lookup(self, table, exchange, routing_key, default):
-        return [queue for rkey, _, queue in table
-                if rkey == routing_key]
+        # Using an OrderedDict to purge queue duplicates while
```

Comment:
I don't know whether keeping the delivery order is meaningful or not, but discarding it leads some other tests to failure. A simpler solution that avoids using an `collections.OrderedDict` (but *randomizes* the result of queues in the method result) would be:
```python
return list(
    set(
        queue for rkey, _, queue in table
        if rkey == routing_key
    )
)
```